### PR TITLE
[macOS] fix layout problems after the parent height change

### LINKED ISSUE #4854: [macOS] Visual glitch when exiting the full screen with ScrollViewer
### Description
The issue doesn't appear with every XAML page so I have submitted the sample.
### Steps to Reproduce

1. Run the app in the sample
2. Enter the full screen
3. Exit the full screen

### Expected Behavior
The app should return to the previous state
### Actual Behavior
The view is scrolled up so part of the view is not visible, while the bottom that should go to the window edge is moved above
### Basic Information

- Version with issue: 4.0.0.8055pre1
- Last known good version:
- IDE:
- Platform Target Frameworks: 
  -macOS Mojave 10.14.2 (18C54)
- Nuget Packages:
- Affected Devices:Mac mini (Late 2014)

### Reproduction Link
[FullScreenProblem.zip](https://github.com/xamarin/Xamarin.Forms/files/2707626/FullScreenProblem.zip)

### LINKED ISSUE #5890: [macOS] Control with VerticalOption=Start is sticked to bottom
### Description

A control with VerticalOption=Start has wrong behaviour during resize. When I change the height of the main window, then items are moved as they were sticked to the bottom of the window. When I change the width of the window, then the position of controls is recalculated and everything looks good.

### Steps to Reproduce

1. Run the attached sample
2. Change the main window height (do not change width)
3. Items are moved, but should be sticked to the top of the window

### Expected Behavior

Items should be sticked to the top of the window.

### Actual Behavior

Items are moved.

### Basic Information

- Version with issue: 4.0.1.268567-nightly
- Last known good version:
- IDE: Visual Studio for Mac 7.8.3
- Platform Target Frameworks: 
  - macOS: 5.2.1.16
- Affected Devices: macOS Mojave (10.14.4)

### Screenshots

After start
start

After height change
height_plus
height_minus

After width change items position is recalculated
height_minus_width_change

### Reproduction Link

[TestAppVerticalOptionStartBug.zip](https://github.com/xamarin/Xamarin.Forms/files/3068807/TestAppVerticalOptionStartBug.zip)

### PR DESCRIPTION
**Pull request by slakul**
### Description of Change ###

The coordinate system in macOS is different than in Xamarin.Forms - the point (X=0, Y=0) is located in the left bottom corner. This fix resolves the problem of moving up the child control when the height of the parent is changed, but size and location of the child is the same. It was necessary to add new properties described below to be able to monitor parent height change.

### Issues Resolved ### 

- fixes #5890 
- fixes #4854

### API Changes ###

Added in Xamarin.Forms.VisualElement (as private):
 - static readonly BindableProperty ParentHeightPropertyForMacOS { get; }
 - double ParentHeightForMacOS { get; set; }

### Platforms Affected ### 

- macOS

### Behavioral/Visual Changes ###

Child controls should not move as described in the resolved issues.

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

Manual tests included in Xamarin.Forms.Controls.Issues:
- Issue4854
- Issue5890

### PR Checklist ###

- [ ] Has automated tests 
- [X] Rebased on top of the target branch at time of PR
- [X] Changes adhere to coding standard

### CODE REVIEW

**Comment by slakul** [comment]
@rmarinho Could you suggest where to put the definition of the attached BP?
On the other hand I can make BP private, and write its name as string in "HandlePropertyChanged" in Xamarin.Forms.Platform.iOS/VisualElementTracker.cs.

File: Xamarin.Forms.Core/VisualElement.cs [hunk 1]
```diff
@@ -43,6 +43,17 @@ public partial class VisualElement : NavigableElement, IAnimatable, IVisualEleme
...
 
+		public static readonly BindableProperty ParentHeightProperty = BindableProperty.Create("ParentHeight", typeof(double), typeof(VisualElement), default(double));
+
+		double ParentHeight
+		{
+			get { return (double)GetValue(ParentHeightProperty); }
+			set { SetValue(ParentHeightProperty, value); }
+		}
+
+		static bool IsMacOsPlatform = Device.RuntimePlatform == Device.macOS;
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/VisualElement.cs]: no

File: Xamarin.Forms.Core/VisualElement.cs [hunk 2]
```diff
@@ -43,6 +43,17 @@ public partial class VisualElement : NavigableElement, IAnimatable, IVisualEleme
 
 		public static readonly BindableProperty HeightProperty = HeightPropertyKey.BindableProperty;
 
+		public static readonly BindableProperty ParentHeightProperty = BindableProperty.Create("ParentHeight", typeof(double), typeof(VisualElement), default(double));
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/VisualElement.cs]: no

**Comment by slakul** [comment]
@StephaneDelcroix @rmarinho In the new commit I made changes regarding to the requested changes.

**Comment by samhouts** [comment]
Thank you for your contribution! We believe that this is a worthy problem to solve, but we think it can be achieved in a different way. We will discuss it and put our findings in the attached issues.

**Pull request closed without merging**